src/app/clusters/general-diagnostics-server/GenericFaultTestEventTriggerDelegate.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "GenericFaultTestEventTriggerDelegate.h"
+
+#include <app/clusters/general-diagnostics-server/general-diagnostics-server.h>
+#include <lib/support/CodeUtils.h>
+
+using namespace ::chip::DeviceLayer;
+
+namespace chip {
+
+bool GenericFaultTestEventTriggerDelegate::DoesEnableKeyMatch(const ByteSpan & enableKey) const
+{
+    return !mEnableKey.empty() && mEnableKey.data_equal(enableKey);
+}
+
+CHIP_ERROR GenericFaultTestEventTriggerDelegate::HandleEventTrigger(uint64_t eventTrigger)
```

Comment:
The eventTrigger must be present and processed.

Please use any value in the 0xFFFF_FFFF_xxxx_yyyy range. Preferably 0xFFFF_FFFF_<VID_IN_HEX>_yyyy.

That will avoid clashes with other vent trigger handler.

You can use the `yyyy` portion to drive different behaviors of test event triggers.

Reply:
OK, agree.
Small comment on event handling - since there is only one delegate could be registered for test events for now, i.e.:
`initParams.testEventTriggerDelegate = &testEventTriggerDelegate;`
For example, we need different event handlers related to different clusters (with dirrerent eventTrigger ids). Should not per-id "routing" ideally happen somewhere else outside that specifiec single handler?